src/main/java/dev/dfonline/codeclient/mixin/world/block/MBlock.java

```diff
@@ -42,4 +67,55 @@ private void getPickStack(WorldView world, BlockPos pos, BlockState state, Callb
                 }
         }
     }
+
+    @Unique
+    private static final Set<String> CALLABLE_BLOCK_NAMES = Set.of(
+            "CALL FUNCTION",
+            "START PROCESS",
+            "FUNCTION",
+            "PROCESS"
+    );
+
+    /**
+     * Will have changed the return value to an appropriate string if it returns true
+     * @param pos Position of the block to try and copy function/process name from
+     * @return Null if a name can't be copied from that position, else the ItemStack of the appropriate DF string varitem
+     */
+    @Unique
+    private ItemStack tryPickFunctionName(BlockPos pos) {
+        if (CodeClient.MC.world.getBlockEntity(pos) instanceof SignBlockEntity signBlock) {
+            try {
+                SignText text = signBlock.getFrontText();
+
+                String firstLine = text.getMessage(0, false).getString();
+
+                if (!CALLABLE_BLOCK_NAMES.contains(firstLine))
+                    return null;
+
+                DFItem string = new DFItem(new ItemStack(Items.STRING, 1));
+
+                String funcName = text.getMessage(1, false).getString();
+
+                string.setName(
+                        // stop it from being italicized like anvil renaming
+                        Text.literal(funcName).setStyle(Style.EMPTY.withItalic(false))
+                    );
+
```

Comment:
what's with these newlines?

Reply:
I've squished some things into fewer lines now, if that's what you meant :3